plugins/pipelines/test/models/stage_test.rb

```diff
@@ -0,0 +1,23 @@
+require_relative '../test_helper'
+
+describe Stage do
+  describe '#next_stage' do
+    let(:project) { Project.new(name: 'foo') }
+    let(:stage1) { Stage.new(project: project, name: 'stage1') }
+    let(:stage2) { Stage.new(project: project, name: 'stage2') }
+    let(:stage3) { Stage.new(project: project, name: 'stage3') }
+
+    before do
+      project.stages = [stage1, stage2, stage3]
+    end
+
+    it 'should return next created stage if no pipeline set' do
```

Comment:
There is the button on the deploy page that showed a naive 'next stage' before this feature.